kickstarter/android-oss
# WEB-670:   [Android] Add Meta CAPI tracking support to project, rewards, payments and thanks pages.

# 📲 What

Add Meta CAPI tracking support to project, rewards, payments and thanks pages.

# 🤔 Why

to give creators insight into their paid marketing spend and provide the ability to optimise ad spend for conversions

# 🛠 How
Create SendCAPIEventUseCase to call graph.ql mutation
```
 private val canSendCAPIEventFlag = (
        optimizely.isFeatureEnabled(OptimizelyFeature.Key.ANDROID_CONSENT_MANAGEMENT) &&
            sharedPreferences.getBoolean(SharedPreferenceKey.CONSENT_MANAGEMENT_PREFERENCE, false) &&
            optimizely.isFeatureEnabled(OptimizelyFeature.Key.ANDROID_CAPI_INTEGRATION)
        )
```
add events to view Project, rewards, payments and thanks pages.


# 📋 QA

- Change in the `ApplicationModule` the retrofit `interceptor.setLevel(HttpLoggingInterceptor.Level.HEADERS);` to     `interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);`

- We've been using this [project](https://staging.kickstarter.com/projects/ionbp/ion-infinity-fountain-pen?ref=profile_saved_projects_ended) for testing. It has its sendMetaCapiEvents property set to true.

- Check that features flags enabled for device 

-Check logcat for logging CAPI HTTP event  
![image](https://user-images.githubusercontent.com/1075310/224176151-83ea416f-b249-4f9f-9695-5ed0234f22dd.png)

# Story 📖
https://kickstarter.atlassian.net/browse/WEB-670
